src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractTranslator.java

```diff
@@ -466,6 +510,14 @@ public String translate(final Case aCase) {
      */
     public abstract String translate(DbColumn dc);
 
+    /**
+     * Translates {@link DbColumnType}.
+     *
+     * @param type The object to translate.
+     * @return The string representation of the given object.
+     */
+    public abstract String translate(DbColumnType type);
```

Comment:
This was not what I had in my mind.  
I don't believe this translate method makes sense.

I was expecting that the contents of this method was inside each `translate(Cast)` directly.  
I believe this is better because it may be misleading because the `translate(DbColumn)` has some specific types of the CAST operation.

What do you think?

Reply:
I understand your point. It makes sense.
I'm not sure if we should make that switch-case method a private one, for each class, or have the code in the cast translate itself.
The first allows for shorter methods, but I believe it will impact readability because the exception is not visible.

What do you think?